homeassistant/components/stream/core.py

```diff
@@ -59,34 +62,34 @@ class IdleTimer:
     """
 
     def __init__(
-        self, hass: HomeAssistant, timeout: int, idle_callback: Callable[[], None]
+        self, hass: HomeAssistant, timeout: int, idle_callback: CALLBACK_TYPE
     ) -> None:
         """Initialize IdleTimer."""
         self._hass = hass
         self._timeout = timeout
         self._callback = idle_callback
-        self._unsub = None
+        self._unsub: CALLBACK_TYPE | None = None
         self.idle = False
 
-    def start(self):
+    def start(self) -> None:
         """Start the idle timer if not already started."""
         self.idle = False
         if self._unsub is None:
             self._unsub = async_call_later(self._hass, self._timeout, self.fire)
 
-    def awake(self):
+    def awake(self) -> None:
         """Keep the idle time alive by resetting the timeout."""
         self.idle = False
         # Reset idle timeout
         self.clear()
         self._unsub = async_call_later(self._hass, self._timeout, self.fire)
 
-    def clear(self):
+    def clear(self) -> None:
         """Clear and disable the timer if it has not already fired."""
         if self._unsub is not None:
             self._unsub()
 
-    def fire(self, _now=None):
+    def fire(self, _now: Any = None) -> None:
```

Comment:
Not sure what type _now should be. We don't use it.

Reply:
```suggestion
    def fire(self, _now: datetime.datetime) -> None:
```

Reply:
Is there a better type to use? It seems like the parameter is only there to satisfy the signature having an argument and nothing is actually passed. Seems like we could just as easily use None as the type

Reply:
Well, if `datetime` is passed here, it's better to specify that instead of other type. It passed from:
https://github.com/home-assistant/core/blob/a989677bef1a79b66a017c4643973c735989f9cf/homeassistant/helpers/event.py#L1195

Reply:
Oh, I see it now, thanks.